prql-compiler/prqlc/src/cli.rs

```diff
@@ -398,14 +398,15 @@ mod clio_extended {
             if path == "-" {
                 Ok(Self::std())
             } else {
-                let file = File::open(path)?;
-                if file.metadata()?.is_dir() {
-                    return Ok(Input::Directory(path.to_os_string()));
+                let pathbuf = PathBuf::from(path);
+                if pathbuf.is_dir() {
```

Comment:
The difference is here — `File::open(path)` on a directory on Windows seems to panic, but is fine on Mac & Linux...

Reply:
Oh god, thank you!

I was just about to drop support for Windows. Debugging in GH actions is hell and dealing with a bug for Windows specifically is nerve wracking.

Thank you again.